### ISSUE TITLE
Support intersection types

### ISSUE THREAD

**Issue by chancancode** [gajus/eslint-plugin-jsdoc#545]
## Motivation

When using TypeScript with JSDoc, the types can be (almost) anything TypeScript supports. This seems to be mostly true already, however, the following doesn't work and give a syntax error:

```js
/**
 * @typedef {object} Foo
 * @property {boolean} isFoo
 *
 * @typedef {object} Bar
 * @property {boolean} isBar
 *
 * @typedef {Foo & Bar} FooAndBar
 *           ~~~~~~~~~ Syntax error in type: Foo & Bar  jsdoc/valid-types
 * 
 * @typedef {Foo | Bar} FooOrBar
 *           ~~~~~~~~~ Ok: union types works
 */
```

This is not as simple as disabling `jsdoc/valid-types`. Fundamentally, if this ESLint plugin doesn't understand the intersection type syntax, it will do other things incorrectly. For example, it wouldn't be able to error when `Foo` or `Bar` doesn't refer to valid types (e.g. typos), and it can't mark the `Foo` and `Bar` values as used to disable the `no-unused-vars` warning from ESLint.

## Current behavior

Syntax error on intersection types `X & Y`.

## Desired behavior

No syntax error on intersection types `X & Y`, and it understand both `X` and `Y` here are type symbols.

## Alternatives considered

???

**Comment by chancancode** [comment]
I understand this is not an official JSDoc feature, so it would have to be opt-in. But it would be nice if there is a configuration option in the parser to allow this.

**Comment by brettz9** [comment]
Good write-up, thank you. This should become available if the intersection types component of https://github.com/jsdoctypeparser/jsdoctypeparser/issues/50#issue-397917278 can be implemented. You may subscribe there (we tend to update the dependencies of `eslint-plugin-jsdoc` fairly frequently, but if that issue gets merged and we need a reminder, feel free to file a new issue; I'd like to avoid tracking type-specific issues if possible). Thanks again!

**Comment by chancancode** [comment]
Thank you for the pointer and the quick response!

**Comment by brettz9** [comment]
This should now be working as of 96dcdced66c9a8cfab09387dd76b752cb79d0718 (v29.0.0 -- though latest is now at v29.1.0)
